numtraits.py

```diff
@@ -38,16 +38,23 @@
 class NumericalTrait(TraitType):
     info_text = 'a numerical trait, either a scalar or a vector'
     def __init__(self, ndim=None, shape=None, domain=None,
-                 default=None, convertible_to=None):
-        super(NumericalTrait, self).__init__()
+                 default=None, convertible_to=None, nullable=False):
+        # We use the default value for the traitlet in the following cases:
+        # - default is not None,
+        # - default is None and the trait is nullable.
+        # In the other cases, we assume that a None default means that
+        # there is no default value defined.
+        if (default is None and nullable) or default is not None:
+            super(NumericalTrait, self).__init__(default_value=default)
+        else:
+            super(NumericalTrait, self).__init__()
```

Comment:
What does the ``default_value`` argument default to? Can you have just one ``super`` call and adapt the value of ``default`` appropriately?